DSpace/DSpace
# Removed timestamp from dc.date.issued when importing from external sources

## References
* Fixes #8744

## Description
Removed the timestamp from the dates when importing from arXiv and made a change on how the date converter worked for Pubmed to always display the imported date in the format the matched.

## Instructions for Reviewers
List of changes in this PR:
* Made a change to the arXiv metadata mapping file (`arxiv-integration.xml`) to import the date in the `yyyy-MM-dd` format
* Made some changes to `PubmedDateMetadatumContributor.java` to render the date in the same format as the first format from `dateFormatsToAttempt` that matches. This change needed to be done because otherwise it always automatically filled in the day and month of the `dc.date.issued`. (so when the date was fore example `2021` it would automatically change it to `2021-01-01`)
* Added a new `dateFormatsToAttempt` with only the year as fallback if both the month and day are missing
* Removed the `metadataFieldMapping.toDCValue` as it changes the date to use the timezones, which can lead to incorrect dates if your in a + timezone like for example Belgium (which is +1) the date is also always subtracts that amount of hours which can lead to dates who are incorrect (`2021` will become `2020-12-31T23:00:00Z` which means that it will be saved as `2020`)

**Guidance for how to test and review this PR:**
- Go to MyDSpace > Import from External Sources
- Search for "[2301.00010](https://export.arxiv.org/api/query?search_query=2301.00010)" and select `arXiv` as external source. When you click on import the published date should not contain the timestamp anymore
- Search for "[32223729](https://pubmed2xl.com/xml/)" and check that items with only the year as `PubDate` also have their date mapped to `dc.date.issued`. Also check that only the year is imported and that the month and day are not set

## Checklist
